DQM/L1TMonitor/src/L1TCSCTF.cc

```diff
@@ -544,46 +787,159 @@ void L1TCSCTF::analyze(const Event& e, const EventSetup& c)
-              } catch(cms::Exception &) { 
-                bzero(&gblPhi,sizeof(gblPhi)); 
+                gblPhi = srLUTs_[fpga]->globalPhiME(lclPhi.phi_local, lct->getKeyWG(), cscId+1, gangedME11a_);
+              } catch(cms::Exception &) {
+                bzero(&gblPhi,sizeof(gblPhi));
               }
-				
+
               gbletadat gblEta;
               try {
-                gblEta = srLUTs_[fpga]->globalEtaME(lclPhi.phi_bend_local, lclPhi.phi_local, lct->getKeyWG(), cscId+1);
-              } catch(cms::Exception &) { 
-                bzero(&gblEta,sizeof(gblEta)); 
+                gblEta = srLUTs_[fpga]->globalEtaME(lclPhi.phi_bend_local, lclPhi.phi_local, lct->getKeyWG(), cscId+1, gangedME11a_);
+              } catch(cms::Exception &) {
+                bzero(&gblEta,sizeof(gblEta));
+              }
+
+
+              //TrackStub
+              csctf::TrackStub theStub((*lct), (*csc).first);
+              theStub.setPhiPacked(gblPhi.global_phi);
+              theStub.setEtaPacked(gblEta.global_eta);
+
+              float etaG = theStub.etaValue();
+              float phiG = fmod( theStub.phiValue()+15.0*M_PI/180+(sector)*60.0*M_PI/180, 2.*M_PI );
+
+
+              // BX plots
+              // minus side
+              if (endcap == 1 && station == 0 && ring == 1) { csctflcts -> Fill(bx, 8.5); }
+              if (endcap == 1 && station == 0 && ring == 2) { csctflcts -> Fill(bx, 7.5); }
+              if (endcap == 1 && station == 0 && ring == 3) { csctflcts -> Fill(bx, 6.5); }
+              if (endcap == 1 && station == 1 && ring == 1) { csctflcts -> Fill(bx, 5.5); }
+              if (endcap == 1 && station == 1 && ring == 2) { csctflcts -> Fill(bx, 4.5); }
+              if (endcap == 1 && station == 2 && ring == 1) { csctflcts -> Fill(bx, 3.5); }
+              if (endcap == 1 && station == 2 && ring == 2) { csctflcts -> Fill(bx, 2.5); }
+              if (endcap == 1 && station == 3 && ring == 1) { csctflcts -> Fill(bx, 1.5); }
+              if (endcap == 1 && station == 3 && ring == 2) { csctflcts -> Fill(bx, 0.5); }
+
+              // plus side
+              if (endcap == 0 && station == 0 && ring == 1) { csctflcts -> Fill(bx, 9.5); }
+              if (endcap == 0 && station == 0 && ring == 2) { csctflcts -> Fill(bx, 10.5); }
+              if (endcap == 0 && station == 0 && ring == 3) { csctflcts -> Fill(bx, 11.5); }
+              if (endcap == 0 && station == 1 && ring == 1) { csctflcts -> Fill(bx, 12.5); }
+              if (endcap == 0 && station == 1 && ring == 2) { csctflcts -> Fill(bx, 13.5); }
+              if (endcap == 0 && station == 2 && ring == 1) { csctflcts -> Fill(bx, 14.5); }
+              if (endcap == 0 && station == 2 && ring == 2) { csctflcts -> Fill(bx, 15.5); }
+              if (endcap == 0 && station == 3 && ring == 1) { csctflcts -> Fill(bx, 16.5); }
+              if (endcap == 0 && station == 3 && ring == 2) { csctflcts -> Fill(bx, 17.5); }
+
```

Comment:
i would rather suggest using comments to make the assignments clear - having one if statement per Fill command is too many

> On Mar 17, 2015, at 11:17 AM, rjwang <notifications@github.com> wrote:
> 
> In DQM/L1TMonitor/src/L1TCSCTF.cc:
> 
> > +              if (endcap == 1 && station == 2 && ring == 1) { csctflcts -> Fill(bx, 3.5); }
> > +              if (endcap == 1 && station == 2 && ring == 2) { csctflcts -> Fill(bx, 2.5); }
> > +              if (endcap == 1 && station == 3 && ring == 1) { csctflcts -> Fill(bx, 1.5); }
> > +              if (endcap == 1 && station == 3 && ring == 2) { csctflcts -> Fill(bx, 0.5); }
> > +
> > +              // plus side
> > +              if (endcap == 0 && station == 0 && ring == 1) { csctflcts -> Fill(bx, 9.5); }
> > +              if (endcap == 0 && station == 0 && ring == 2) { csctflcts -> Fill(bx, 10.5); }
> > +              if (endcap == 0 && station == 0 && ring == 3) { csctflcts -> Fill(bx, 11.5); }
> > +              if (endcap == 0 && station == 1 && ring == 1) { csctflcts -> Fill(bx, 12.5); }
> > +              if (endcap == 0 && station == 1 && ring == 2) { csctflcts -> Fill(bx, 13.5); }
> > +              if (endcap == 0 && station == 2 && ring == 1) { csctflcts -> Fill(bx, 14.5); }
> > +              if (endcap == 0 && station == 2 && ring == 2) { csctflcts -> Fill(bx, 15.5); }
> > +              if (endcap == 0 && station == 3 && ring == 1) { csctflcts -> Fill(bx, 16.5); }
> > +              if (endcap == 0 && station == 3 && ring == 2) { csctflcts -> Fill(bx, 17.5); }
> > +
> 
> @davidlange6
> 
> Hi David,
> 
> What I mean is like this
> // BX plots
> // minus side
> if (endcap == 1 && station == 0 && ring == 1) { csctflcts -> Fill(bx, 8.5); }
> else if (endcap == 1 && station == 0 && ring == 2) { csctflcts -> Fill(bx, 7.5); }
> else if (endcap == 1 && station == 0 && ring == 3) { csctflcts -> Fill(bx, 6.5); }
> else if (endcap == 1 && station == 1 && ring == 1) { csctflcts -> Fill(bx, 5.5); }
> else if (endcap == 1 && station == 1 && ring == 2) { csctflcts -> Fill(bx, 4.5); }
> else if (endcap == 1 && station == 2 && ring == 1) { csctflcts -> Fill(bx, 3.5); }
> else if (endcap == 1 && station == 2 && ring == 2) { csctflcts -> Fill(bx, 2.5); }
> else if (endcap == 1 && station == 3 && ring == 1) { csctflcts -> Fill(bx, 1.5); }
> else if (endcap == 1 && station == 3 && ring == 2) { csctflcts -> Fill(bx, 0.5); }
> // plus side
> else if (endcap == 0 && station == 0 && ring == 1) { csctflcts -> Fill(bx, 9.5); }
> else if (endcap == 0 && station == 0 && ring == 2) { csctflcts -> Fill(bx, 10.5); }
> else if (endcap == 0 && station == 0 && ring == 3) { csctflcts -> Fill(bx, 11.5); }
> else if (endcap == 0 && station == 1 && ring == 1) { csctflcts -> Fill(bx, 12.5); }
> else if (endcap == 0 && station == 1 && ring == 2) { csctflcts -> Fill(bx, 13.5); }
> else if (endcap == 0 && station == 2 && ring == 1) { csctflcts -> Fill(bx, 14.5); }
> else if (endcap == 0 && station == 2 && ring == 2) { csctflcts -> Fill(bx, 15.5); }
> else if (endcap == 0 && station == 3 && ring == 1) { csctflcts -> Fill(bx, 16.5); }
> else if (endcap == 0 && station == 3 && ring == 2) { csctflcts -> Fill(bx, 17.5); }
> 
> if we do deeper nesting, like
> if (endcap==0) { if (station == 1) { if (ring == 1) ....
> 
> it will be not very clear to see the endcap, station and ring with different values will be assigned to different chambers, it is not good for other people reading and debug.... Do you agree?
> 
> —
> Reply to this email directly or view it on GitHub.
>